content/design/vendoring.md

```diff
@@ -53,3 +53,224 @@ already written):
+> $ govendor add +e         # pull needed packages in
+> ```
+>
+> And you're set. Everyone in your team will be able to deterministically
+> regenerate your code if they tweak the design, without version conflicts.
+
+Instead of using `goagen` to get started, you can write code that'll
+do the same, but as it will mark import paths in code, your prefered
+vendoring tool will be able to pick up any dependencies by following
+the dependency graph of the code:
+
+```
+cd $GOPATH/src/github.com/your/pkg
+mkdir gen
+```
+
+Then put this content into `gen/main.go`:
+
+```
+package main
+
+import (
+	_ "github.com/your/pkg/design"
+
+	"github.com/goadesign/goa/design"
+	"github.com/goadesign/goa/goagen/codegen"
+	genapp "github.com/goadesign/goa/goagen/gen_app"
+	genclient "github.com/goadesign/goa/goagen/gen_client"
+	genjs "github.com/goadesign/goa/goagen/gen_js"
+	genmain "github.com/goadesign/goa/goagen/gen_main"
+	genschema "github.com/goadesign/goa/goagen/gen_schema"
+	genswagger "github.com/goadesign/goa/goagen/gen_swagger"
+)
+
+func main() {
+	codegen.ParseDSL()
+	codegen.Run(
+		&genmain.Generator{
+			API:    design.Design,
+			Target: "app",
+		},
+		&genswagger.Generator{
+			API: design.Design,
+		},
+		&genapp.Generator{
+			API:    design.Design,
+			OutDir: "app",
+			Target: "app",
+			NoTest: true,
+		},
+		&genclient.Generator{
+			API: design.Design,
+		},
+		&genschema.Generator{
+			API: design.Design,
+		},
+		&genjs.Generator{
+			API: design.Design,
+		},
+	)
+}
+```
+
+Remove the generators you don't need from the `Run()` call.
+
+Merely by having that little code in place, all the `goa` code
+generation code will be vendored in.
+
+With `govendor` (you can use whatever vendoring tool you want in a
+similar way), you can simply do this:
+
+```
+$ govendor init
+$ govendor list
+ e  github.com/dimfeld/httppath
+ e  github.com/goadesign/goa/design
+ e  github.com/goadesign/goa/design/apidsl
+ e  github.com/goadesign/goa/dslengine
+ e  github.com/goadesign/goa/goagen/codegen
+ e  github.com/goadesign/goa/goagen/gen_app
+ e  github.com/goadesign/goa/goagen/gen_client
+ e  github.com/goadesign/goa/goagen/gen_js
+ e  github.com/goadesign/goa/goagen/gen_main
+ e  github.com/goadesign/goa/goagen/gen_schema
+ e  github.com/goadesign/goa/goagen/gen_swagger
+ e  github.com/goadesign/goa/goagen/utils
+ e  github.com/goadesign/goa/version
+ e  github.com/manveru/faker
+ e  github.com/satori/go.uuid
+ e  github.com/zach-klippenstein/goregen
+ e  golang.org/x/tools/go/ast/astutil
+ e  gopkg.in/yaml.v2
+ l  github.com/your/pkg/design
+pl  github.com/your/pkg/gen
+```
+
+and another quick run will vendor all you need to generate your code next time:
+
+```
+$ govendor add +ext
+$ wc -l vendor/vendor.json
+181 vendor/vendor.json
+```
+
+This gives you assurance that the code generation libraries and
+templates will only change, for a given project, when you decide to
+explicitly do so, using your vendoring tool.
```

Comment:
This gives you control over when to change the code generation packages using your vendoring tool of choice.